### PR TITLE
feat(aci): use feature check for aci

### PR DESCRIPTION
**Pull request by arturfrombox**
This pr introduces changes to cover ACI codebase with feature checking

### CODE REVIEW

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 1]
```diff
@@ -192,8 +194,11 @@ class DocBaseViewer extends BaseViewer {
         }
 
         this.updateDiscoverabilityResinTag();
-        if (this.options.advancedContentInsights) {
-            this.pageTracker = new PageTracker(this.options.advancedContentInsights, this.options.file);
+
+        const advancedContentInsightsConfig = getFeatureConfig(this.options.features, 'advancedContentInsights');
```
Comment (ahorowitz123) [inline src/lib/viewers/doc/DocBaseViewer.js:198]: Same comment as above about `isFeatureEnabled`

File: src/lib/Preview.js [hunk 2]
```diff
@@ -1845,9 +1851,13 @@ class Preview extends EventEmitter {
...
     updateContentInsightsOptions(options) {
+        this.options.features = { ...this.options.features, advancedContentInsights: options };
         if (this.viewer && this.viewer.pageTracker) {
-            this.previewOptions = { ...this.previewOptions, contentInsights: options };
-            this.viewer.pageTracker.updateOptions(this.previewOptions.contentInsights);
+            this.previewOptions = {
+                ...this.previewOptions,
+                features: { ...this.previewOptions.features, advancedContentInsights: options },
+            };
+            this.viewer.pageTracker.updateOptions(this.previewOptions.features.advancedContentInsights);
```
Comment (ahorowitz123) [inline src/lib/Preview.js:1860]: Can't we just use `options` as the argument to the function here?

**Review by jstoffan — COMMENTED** [review]
What's benefit of these changes if we already have a configuration object for these options? We appear to be increasing complexity at every layer.

File: src/lib/Preview.js (hunk 2 above)
Comment (arturfrombox) [reply src/lib/Preview.js:1860]: can you explain why we should do so ?

**Comment by arturfrombox** [comment]
> What's benefit of these changes if we already have a configuration object for these options? We appear to be increasing complexity at every layer.

It was one of PMAI and intention behind this change was to stick to "feature checking" mechanism implemented for Preview and to hide rest of ACI codebase under the feature flip.

**Comment by arturfrombox** [comment]
@ahorowitz123 @jstoffan  thanks for your comments. I updated this PR

File: src/lib/Preview.js (hunk 2 above)
Comment (loonskai) [reply src/lib/Preview.js:1856]: nitpick: I also found it a bit difficult to grasp having both `options` and `this.options`

Comment (ahorowitz123) [reply src/lib/Preview.js:1856]: I still find this confusing as well. Why not just call `this.viewer.pageTracker.updateOptions(newOptions)` here? It's much easier to read in my option and stays more consistent with the rest of this code block

**Review by ahorowitz123 — APPROVED** [review]

**Review by loonskai — APPROVED** [review]

**Review by ahorowitz123 — APPROVED** [review]

**Pull request merged**